### ISSUE TITLE
E3033 on Lambda FunctionName is incorrect

### ISSUE THREAD

**Issue by dcendents** [aws-cloudformation/cfn-lint#4447]
### CloudFormation Lint Version

1.48.1

### What operating system are you using?

Ubuntu

### Describe the bug

As per the documentation, the Lambda FunctionName must be up to 64 characters: https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/aws-resource-lambda-function.html#cfn-lambda-function-functionname

However `cfn-lint` will accept values up to 140 characters before returning an error

### Expected behavior

Expecting to validate the correct limit of 64 characters.

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
Transform: 'AWS::LanguageExtensions'
Resources:
  LambdaFAToken:
    Type: AWS::Lambda::Function
    Properties:
      FunctionName: very-long-name-exceeding-the-limit-of-64-characters-as-per-documentation
      # FunctionName: however-it-will-fail-with-this-name-that-exceeds-140-characters-but-is-wrong-very-long-name-exceeding-the-limit-of-64-characters-as-per-documentation
      Code:
        ZipFile: |
            exports.handler =  async (event) => {
              const payload = {
                date: new Date(),
                message: 'Hello Lambda CLI World'
              };
              return JSON.stringify(payload);
            };
      Handler: index.handler"
      Runtime: nodejs22.x
      Role: arn:aws:iam::123456789012:role/role-name
```
